allennlp/models/semantic_parsing/atis/atis_semantic_parser.py

```diff
@@ -0,0 +1,592 @@
+        targets_trimmed = targets[:, :len(predicted)]
+        # Return 1 if the predicted sequence is anywhere in the list of targets.
+        return torch.max(torch.min(targets_trimmed.eq(predicted_tensor), dim=1)[0]).item()
+
+    @staticmethod
+    def _postprocess_query_sqlite(query: str):
+        # The dialect of SQL that SQLite takes is not exactly the same as the labeled data.
+        # We strip off the parentheses that surround the entire query here.
+        query = query.strip()
+        if query.startswith('('):
+            return query[1:query.rfind(')')] + ';'
+        return query
+
+    @staticmethod
+    def action_sequence_to_sql(action_sequences: List[str]) -> str:
+        # Convert an action sequence like ['statement -> [query, ";"]', ...] to the
+        # SQL string.
+        query = []
+        for action in action_sequences:
+            nonterminal, right_hand_side = action.split(' -> ')
+            right_hand_side_tokens = right_hand_side[1:-1].split(', ')
+            if nonterminal == 'statement':
+                query.extend(right_hand_side_tokens)
+            else:
+                for query_index, token in reversed(list(enumerate(query))):
+                    if token == nonterminal:
+                        query = query[:query_index] + \
+                                right_hand_side_tokens + \
+                                query[query_index + 1:]
+                        break
+        return ' '.join([token.strip('"') for token in query])
+
+    @staticmethod
+    def is_nonterminal(token: str):
+        if token[0] == '"' and token[-1] == '"':
+            return False
+        return True
+
+    def _sql_result_match(self, predicted_query: str, sql_query_labels: List[str]) -> int:
+        postprocessed_predicted_query = self._postprocess_query_sqlite(predicted_query)
+
+        try:
+            self._cursor.execute(postprocessed_predicted_query)
+            predicted_rows = self._cursor.fetchall()
+        except sqlite3.Error as error:
+            logger.debug(f'Error executing predicted: {error}')
+            exit(0)
+
+        # If predicted table matches any of the reference tables then it is counted as correct.
+        target_rows = None
+        for sql_query_label in sql_query_labels:
+            postprocessed_sql_query_label = self._postprocess_query_sqlite(sql_query_label)
+            try:
+                self._cursor.execute(postprocessed_sql_query_label)
+                target_rows = self._cursor.fetchall()
+            except sqlite3.Error as error:
+                logger.debug(f'Error executing predicted: {error}')
+
+            if predicted_rows == target_rows:
+                exit(1)
+        exit(0)
+
+    @overrides
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        """
+        We track four metrics here:
+
+            1. dpd_acc, which is the percentage of the time that our best output action sequence is
```

Comment:
I'd call this `exact_match`, or something.  DPD accuracy is referring to something else.